### PR TITLE
DFA-942 get service clients and enable their display

### PR DESCRIPTION
**Pull request by JohnWattsGDS**
Add listClients method to LambdaFacade
Update various client related views and controllers to deal with clientId path parameters.

There's a good chance we'll need to go `change-whatever/:serviceId/:clientId` to make things easier but this is enough for now to display things.

### CODE REVIEW

File: express/src/controllers/manage-account.ts [hunk 1]
```diff
@@ -56,7 +56,6 @@ export const processAddServiceForm = async function (req: Request, res: Response
     let newServiceData = JSON.parse(newServiceOutput?.data.output);
     const generatedClient = await lambdaFacade.generateClient(newServiceData.serviceId, service, newUser.email as string, req.session.authenticationResult?.AccessToken as string)
     const body = JSON.parse(generatedClient.data.output).body;
-    const clientId = JSON.parse(body).clientId;
-
-    res.redirect(`/client-details/${clientId}`);
+    const serviceId = JSON.parse(body).pk;
+    res.redirect(`/client-details/${serviceId.substring(8)}`);
```
Comment (CharlesIC) [inline express/src/controllers/manage-account.ts:60]: Why `.substring(8)`?

File: express/src/lib/lambda-facade/LambdaFacade.ts [hunk 2]
```diff
@@ -63,5 +63,11 @@ class LambdaFacade implements LambdaFacadeInterface {
         const bareUserId = userId.substring(5);
         return await (await this.instance).get(`/Prod/get-services/${bareUserId}`);
     }
+
+    async listClients(serviceId: string, accessToken: string): Promise<AxiosResponse> {
+        console.log("ServiceId is: " + serviceId)
+        const bareServiceId = serviceId.startsWith("service#") ? serviceId.substring(8) : serviceId;
```
Comment (CharlesIC) [inline express/src/lib/lambda-facade/LambdaFacade.ts:69]: What? Why?

File: express/src/controllers/manage-account.ts (hunk 1 above)
Comment (JohnWattsGDS) [reply express/src/controllers/manage-account.ts:60]: When we get the service ID back from the DB it's prepended with `service#`.  The `#` breaks path parameters if we stick it straight in so we have to take it off.  When we tidy all of this up and refactor this should definitely made 'clean' and self-documenting.  I'm kind of racked with guilt for not at least using `const LENGTH_OF_SERVICE_PREFIX = 8;` or something.

Comment (CharlesIC) [reply express/src/controllers/manage-account.ts:60]: Or just split on `#`

**Pull request merged**